From d7bc22a80e451f0d768172712f027cdc71b744f6 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 12 Jul 2019 02:09:45 +0200
Subject: [PATCH] Add support for Flatpak config/data dir variables

Those environment variables are by Flatpak when it sets up its own
environment to make it easier for applications to know where to put
their own files, the path of which might not be relative to $HOME.
---
 paths.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/paths.c b/paths.c
index 98730c8694d0..39d1722e32b6 100644
--- a/paths.c
+++ b/paths.c
@@ -553,6 +553,9 @@ dir_entry *get_bundled_dir_list(char *name, size_t *num_out)
 char const *get_config_dir()
 {
 	static char* confdir;
+	if (!confdir) {
+		confdir = getenv("XDG_CONFIG_HOME");
+	}
 	if (!confdir) {
 		char const *base = get_home_dir();
 		if (base) {
@@ -565,6 +568,9 @@ char const *get_config_dir()
 char const *get_userdata_dir()
 {
 	static char* savedir;
+	if (!savedir) {
+		savedir = getenv("XDG_DATA_HOME");
+	}
 	if (!savedir) {
 		char const *base = get_home_dir();
 		if (base) {
-- 
2.55.0

